plugins/gateway/tyk/v3.2.2/plugin.go

```diff
@@ -67,7 +69,11 @@ func PostGetAPIDefinition(_ http.ResponseWriter, r *http.Request) {
 		logger.Error("Failed to get api definition")
 		return
 	}
+	// set the apiDefinition since we dont get it in the response phase
 	ctx.SetDefinition(r, apiDefinition)
+
+	// set request time header
+	r.Header.Set(HTTPXRequestTimeHeader, time.Now().Format(time.RFC3339Nano))
```

Comment:
maybe we can add it to the metadata on the session instead of adding headers (and potentially fail applications):
GetSession/SetSession:
MetaData                map[string]interface{} `json:"meta_data" msg:"meta_data"`

Reply:
this is a great idea to add it to the session metadata